app/models/podcast_import.rb

```diff
@@ -47,13 +49,21 @@ def episode_importing_count
     feed_episode_count - episode_import_placeholders.count
   end
 
+  def remaining_import_count
+    if feed_episode_count.present?
+      feed_episode_count - episode_imports.finished.count
+    end
+  end
+
   def status
     return super unless episode_imports.count > 0
     return super if episode_importing_count > episode_imports.count
 
     if complete?
+      status_complete!
```

Comment:
ok, so this is related to [this thread](https://prx.slack.com/archives/G0FKLK2Q5/p1689705702150919) where nothing sets the import status as complete/failed at the end of an import. I tried to incorporate an `update_status` method into a hook, but i couldn't figure out what the proper end trigger was. `after_save` `if: :status_changed?` didn't seem to trigger the status to update, also tried `if: :finished?` but that wasn't correct either. so in the meantime i put the status update here, which is admittedly still not the correct place for it.

would placing the status update [after this](https://github.com/PRX/feeder.prx.org/blob/05d5462625e84778fc7a610a36acfb80b1b0e82a/app/models/podcast_import.rb#L115) be the right way to go? but i'm not sure if that method waits for the worker to finish the jobs before moving on? want to avoid the status reading `complete` but the imports are still happening.

Reply:
Huh... well maybe let's just open a ticket to sort this out?  Seems a bit complex to iron out entirely here.

The `import_episodes!` method won't actually work for this.  It does create all the `EpisodeImport`s, but it doesn't process them.  It kicks off another async job for each of them in `enqueue_episode_import_jobs`.

So as each of the N `EpisodeImport`s change status (async), we need to re-calculate the `PodcastImport` status.

But yeah, let's open a ticket.